spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java

```diff
@@ -35,17 +38,17 @@
 @ConditionalOnClass(CloudStorageAccount.class)
 @ConditionalOnProperty(name = "spring.cloud.azure.storage.account")
 @EnableConfigurationProperties(AzureStorageProperties.class)
+@Import(AzureStorageProtocolResolver.class)
```

Comment:
Why Import? Is it being used here?

Reply:
This import means that we need this class registered as bean for later usage. We rely on this resolver to handle azure storage specific url

Reply:
According to https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Import.html, this import also allows for regular component classes, which register the imported class as bean in application context. I have confirmed this, other ways such as `@Component` and configuration class provide such bean don't work neither